### PR TITLE
Add docker support for module sender sample

### PR DESCRIPTION
**Pull request by avranju**
Add Docker support for the edge module sender sample so that we can build an image for this sample and test with Azure IoT Edge.

### CODE REVIEW

**Comment by kiranpradeep** [comment]
Sorry for jumping the gun. But the corresponding C# filter module([1](https://docs.microsoft.com/en-us/azure/iot-edge/tutorial-csharp-module)) takes threshold from module twin. Looking for a C sample which could read from module twin.

**Comment by jspaith** [comment]
@kiranpradeep Please open a github issue to track the request. We'll close the pull request once it gets merged up and I don't want to lose your feedback here. Thanks

**Comment by jspaith** [comment]
Oh yes @kiranpradeep, just to be more clear Raj's scope here is limited to "docker-ification" and limited changes to play nice in Edge which is why i asked for different item tracking your ask.  JS

**Comment by avranju** [comment]
@jspaith For some reason I don't see your review comments on the PR. But since we discussed offline I submitted the changes you requested last night.

**Comment by kiranpradeep** [comment]
@jspaith Thanks. Would like your guidance on which GitHub repo to file these requests under( these are not real issues but more user requests ). Is it iot-edge([1](https://github.com/Azure/iot-edge)), azure-iot-sdk-c([2](https://github.com/Azure/azure-iot-sdk-c)) or iot-edge suggestions page([3](https://feedback.azure.com/forums/907045-azure-iot-edge))? In fact, I was suprised to see docker files going into azure-iot-sdk-c rather than iot-edge.

File: iothub_client/samples/iothub_client_sample_module_sender/iothub_client_sample_module_sender.c [hunk 1]
```diff
@@ -23,13 +23,13 @@
 
 /*String containing Hostname, Device Id & Device Key, ModuleID, and GatewayHostName in the format:                          */
 /*  "HostName=<host_name>;DeviceId=<device_id>;SharedAccessKey=<device_key>;ModuleId=<Module_Id>;GatewayHostName=127.0.0.1" */
-static const char* connectionString = "[device connection string]";
+static char* connectionString = "[device connection string]";
```
Comment (jspaith) [inline iothub_client/samples/iothub_client_sample_module_sender/iothub_client_sample_module_sender.c]: Please leave this as a 'const char*'.  This means the value cannot be changed, but the pointer can, which is what we want.

File: iothub_client/samples/iothub_client_sample_module_sender/iothub_client_sample_module_sender.c [hunk 2]
```diff
@@ -127,9 +127,14 @@ static void SendConfirmationCallback(IOTHUB_CLIENT_CONFIRMATION_RESULT result, v
 void iothub_client_sample_module_sender(void)
 {
     IOTHUB_CLIENT_LL_HANDLE iotHubClientHandle;
-
     EVENT_INSTANCE messages[MESSAGE_COUNT];
 
+    char *envCS = getenv("EdgeHubConnectionString");
```
Comment (jspaith) [inline iothub_client/samples/iothub_client_sample_module_sender/iothub_client_sample_module_sender.c]: We tend to not use abbreviations for this, so connectionStringFromEnvironment would be closer.

File: iothub_client/samples/iothub_client_sample_module_sender/start.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
```
Comment (jspaith) [inline iothub_client/samples/iothub_client_sample_module_sender/start.sh]: This should probably also be under the "./linux" directory.  Unless we are tying into the "bash on Windows" subsystem stuff for Win10 or assuming a Cygwin or other type Unix'y environment on Win, this looks pretty Linux.

**Comment by jspaith** [comment]
@avranju - I have a few more tweaks but in general looks good, once you address I can get checkin process going.

**Comment by jspaith** [comment]
@kiranpradeep - for Edge wide suggestions / feedback - either for Edge Hub / Agent etc. or general programming model questions across all SDK's - please continue using one of the existing Edge channels.

If it's a request where the changes are going into the C SDK - e.g. the sample JSON you've made - it's better to go here.  I don't check the Edge so someone has to forward it to me, but I'll see C requests here and be able to react faster.

RE where to checkin the Docker files, historically when I've seen code separated from manifest then they tend to get out of alignment as someone updates one but not the other.  So by keeping Docker setup close to the actual sample code, less likely.  But for this case, if there's something different I'm not seeing I'm open to feedback.

File: iothub_client/samples/iothub_client_sample_module_sender/iothub_client_sample_module_sender.c (hunk 1 above)
Comment (avranju) [reply iothub_client/samples/iothub_client_sample_module_sender/iothub_client_sample_module_sender.c]: Sure.

File: iothub_client/samples/iothub_client_sample_module_sender/iothub_client_sample_module_sender.c (hunk 2 above)
Comment (avranju) [reply iothub_client/samples/iothub_client_sample_module_sender/iothub_client_sample_module_sender.c]: :) I like small variable names but that's me. Changed it to `connectionStringFromEnvironment`.

File: iothub_client/samples/iothub_client_sample_module_sender/start.sh (hunk 3 above)
Comment (avranju) [reply iothub_client/samples/iothub_client_sample_module_sender/start.sh]: Its already in a folder called `docker\linux\amd64`. We should probably make a `Dockerfile` for ARM too. I didn't have the time to setup and test on a Pi.

**Pull request merged**